python-package/lightgbm/sklearn.py

```diff
@@ -785,7 +808,24 @@ def fit(
 
         train_set = Dataset(data=_X, label=_y, weight=sample_weight, group=group,
                             init_score=init_score, categorical_feature=categorical_feature,
-                            params=params)
+                            params=params, free_raw_data=False)
```

Comment:
Why is it necessary to pass `free_raw_data=False` here? That would be a significant breaking change, as it means that these estimators would default to now holding a copy of the raw training data in memory. That increases the likelihood of out-of-memory errors.

Learning that doing that was required to implement automatic early stopping would make me less confident that this feature should be enabled in LightGBM.

Reply:
I have found a workaround for that, `free_raw_data` is not set to `False` anymore.

Reply:
great, thank you